doc/examples/denoise_mppca.py

```diff
@@ -0,0 +1,282 @@
+
+data = img.get_data()
+
+affine = img.affine
+
+"""
+For the sake of simplicity, we only select two non-zero b-values for this
+example.
+"""
+
+bvals = gtab.bvals
+
+bvecs = gtab.bvecs
+
+sel_b = np.logical_or(np.logical_or(bvals == 0, bvals == 1000), bvals == 2000)
+
+data = data[..., sel_b]
+
+gtab = gradient_table(bvals[sel_b], bvecs[sel_b])
+
+print(data.shape)
+
+"""
+As one can see from its shape, the selected data contains a total of 67
+volumes of images corresponding to all the diffusion gradient directions
+of the selected b-values.
+
+The local PCA denoising based on the Marcenko and Pastur distribution
+can be performed by calling the function ``localpca`` without specifying a
+prior estimate to the noise standard deviation:
+"""
+
+t = time()
+
+denoised_arr = localpca(data, sigma=None, patch_radius=2)
+
+print("Time taken for local PCA ", -t + time())
+
+"""
+Internally, the ``localpca`` algorithm locally denoises the
+diffusion-weighted data using a 3D sliding window which is defined by the
+variable patch_radius. In total, this window should comprise a larger number
+of voxels than the number of diffusion-weighted volumes. Since our data has a
+total of 67 volumes, the patch_radius is set to 2 which corresponds to a 5x5x5
+sliding window comprising a total of 125 voxels.
+
+To assess the performance of the Marcenko-Pastur PCA denosing algorithm,
+an axial slice of the original data, denoised data, and residuals are plotted
+below:
+"""
+
+sli = data.shape[2] // 2
+gra = data.shape[3] - 1
+orig = data[:, :, sli, gra]
+den = denoised_arr[:, :, sli, gra]
+rms_diff = np.sqrt((orig - den) ** 2)
+
+fig1, ax = plt.subplots(1, 3, figsize=(12, 6),
+                        subplot_kw={'xticks': [], 'yticks': []})
+
+fig1.subplots_adjust(hspace=0.3, wspace=0.05)
+
+ax.flat[0].imshow(orig.T, cmap='gray', interpolation='none',
+                  origin='lower')
+ax.flat[0].set_title('Original')
+ax.flat[1].imshow(den.T, cmap='gray', interpolation='none',
+                  origin='lower')
+ax.flat[1].set_title('Denoised Output')
+ax.flat[2].imshow(rms_diff.T, cmap='gray', interpolation='none',
+                  origin='lower')
+ax.flat[2].set_title('Residuals')
+
+fig1.savefig('denoised_mppca.png')
+
+print("The result saved in denoised_mppca.png")
+
+"""
+.. figure:: denoised_mppca.png
+   :align: center
+
+The noise suppressing can be visually appreciated by comparing the original
+data slice (left panel) to its denoised version (middle panel). The difference
+between original and denoised data showing only random noise indicates that
+the data's structural information is preserved by the PCA denoising algorithm
+(left panel).
+
+Below we show how the denoised data can be saved.
+"""
+
+nib.save(nib.Nifti1Image(denoised_arr,
+                         affine), 'denoised_mppca.nii.gz')
+
+print("Entire denoised data saved in denoised_mppca.nii.gz")
+
+"""
+Additionally, we show on this example how the PCA denoising algorithm effects
```

Comment:
Additionally, we show **in** this example how the PCA denoising algorithm **affects**
different diffusion measurements.